### DISCUSSION TITLE
Eslint complains about type parameter declaration

### DISCUSSION THREAD

**Discussion by VampishWolf** [wevm/viem#810] [Question] [answered]
I upgraded my wagmi package which brought viem along with it and now when I try to build my project it fails due to eslint errors.

Following is the response from my terminal -

```
./node_modules/viem/node_modules/abitype/dist/types/human-readable/formatAbi.d.ts:18:35
Type error: Type parameter declaration expected.

  16 |  * @returns Human-readable ABI
  17 |  */
> 18 | export declare function formatAbi<const TAbi extends Abi | readonly unknown[]>(abi: TAbi): FormatAbi<TAbi>;
     |                                   ^
  19 | //# sourceMappingURL=formatAbi.d.ts.map
error Command failed with exit code 1.
```

What could be done to solve this? Are some changes needed on viem's end or a eslint configuration issue?

**Comment by fubhy** [comment, marked as answer]
@VampishWolf Make sure you are on TypeScript 5.x.

**Reply by VampishWolf** [reply]
Oh, I was on "typescript": "4.9.4",

**Comment by tmm** [comment]
What eslint version are you using? Support for const type modifiers was added in [5.5.50](https://github.com/typescript-eslint/typescript-eslint/releases/tag/v5.55.0).

- https://github.com/typescript-eslint/typescript-eslint/pull/6600

**Reply by VampishWolf** [reply]
I'm on these versions
"eslint": "^8.43.0",
"eslint-config-next": "13.1.2",
